views/forms/CharFilterForm/index.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react'
+import {
+  Button,
+  Checkbox,
+  List,
+  ListItem,
+  ListItemText,
+} from '@material-ui/core'
+import { FILTER_CATEGORIES } from '../../../constants'
+
+const ChartFilterForm = ({ initialValues, onSubmit, classes }) => {
+  const [values, setValues] = useState(initialValues)
+  const onChange = (target, filterKey) => {
+    setValues({
+      ...values,
+      [filterKey]: values[filterKey].map((filter) =>
+        filter.name === target.name
+          ? { ...filter, value: target.checked }
+          : filter
+      ),
+    })
+  }
+
+  return (
+    <>
+      {Object.keys(values).map((filterKey) => {
+        return (
+          <List
+            id={filterKey}
+            key={filterKey}
+            component="nav"
+            className={classes.filters}
+          >
+            <ListItemText
+              primary={FILTER_CATEGORIES[filterKey]}
+              style={{ paddingLeft: '10px' }}
+              className={classes.filterText}
+            />
+            {values[filterKey].map((filter) => {
+              const filterID = `${filterKey}-${filter.name}`
+
+              return (
+                <List key={filterID} component="div" disablePadding>
+                  <ListItem className={classes.filterNested}>
+                    <ListItemText
+                      htmlFor={filterID}
+                      primary={filter.name}
+                      inset
+                    />
+                    <Checkbox
+                      checked={filter.value}
+                      onChange={({ target }) => onChange(target, filterKey)}
+                      name={filter.name}
+                      id={filterID}
+                      value={filter.value}
+                    />
+                  </ListItem>
+                </List>
+              )
+            })}
+          </List>
+        )
+      })}
+      <Button
+        onClick={() => onSubmit(values)}
```

Comment:
if we make this component a form component we should probably make this a submit button. In that case we would either want `onSubmit(values)` to be called as part of an `onSubmit` prop on the form component or call `event.preventDefault()` on this `onClick` prop.

Reply:
https://github.com/AMP-SCZ/dpdash/pull/248/commits/147bf8a9965b532f34bb176e1e651f339c9bf387